### PR TITLE
User lookup by email

### PR DESCRIPTION
**Pull request by zimnyjakub**
In our project we match users based on their emails in slack enterprise grid and other external systems.
We needed functionality to find slack user id by email, so that we could message people directly based on their email address.

### CODE REVIEW

File: src/Noobot.Core/NoobotCore.cs [hunk 1]
```diff
@@ -252,6 +252,12 @@ public string GetUserIdForUsername(string username)
             return string.IsNullOrEmpty(user.Key) ? string.Empty : user.Key;
         }
 
+        public string GetUserIdForUserEmail(string email)
+        {
+            var user = _connection.UserCache.Where(x => x.Value.Email != null).FirstOrDefault(x => x.Value.Email.Equals(email, StringComparison.OrdinalIgnoreCase));
```
Comment (Workshop2) [inline src/Noobot.Core/NoobotCore.cs]: This is a very long method, could it be broken up?

File: src/Noobot.Core/NoobotCore.cs [hunk 2]
```diff
@@ -252,6 +252,12 @@ public string GetUserIdForUsername(string username)
             return string.IsNullOrEmpty(user.Key) ? string.Empty : user.Key;
         }
 
+        public string GetUserIdForUserEmail(string email)
```
Comment (Workshop2) [inline src/Noobot.Core/NoobotCore.cs:256]: This is missing unit tests

**Review by Workshop2 — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see feedback

**Comment by zimnyjakub** [comment]
Sure, will do

File: src/Noobot.Core/NoobotCore.cs [hunk 3]
```diff
@@ -48,6 +48,7 @@ public NoobotCore(IConfigReader configReader, ILog log, INoobotContainer contain
             _log.Info("Connected!");
             _log.Info($"Bots Name: {_connection.Self.Name}");
             _log.Info($"Team Name: {_connection.Team.Name}");
+            _log.Info($"Id for user: {GetUserIdForUserEmail("zimny.jot@gmail.com")}");
```
Comment (Workshop2) [inline src/Noobot.Core/NoobotCore.cs:51]: Not sure this is meant to be there

**Comment by Workshop2** [comment]
Thanks for your hard work. I can clean up that last bit when I go on a computer

File: src/Noobot.Core/NoobotCore.cs (hunk 3 above)
Comment (zimnyjakub) [reply src/Noobot.Core/NoobotCore.cs:51]: Whoops, you're right

**Comment by Workshop2** [comment]
Released https://github.com/noobot/noobot/releases/tag/3.1.55

**Pull request merged**